cmd/app.go

```diff
@@ -433,16 +433,25 @@ func appUpgrade(ctx *cli.Context) error {
 	}
 
 	answers := app.Answers
-	answers, err = processAnswers(ctx, c, nil, answers, false)
-	if err != nil {
-		return err
+	if ctx.String("answers") != "" {
+		answers, err = processAnswers(ctx, c, nil, answers, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	values := app.ValuesYaml
+	if ctx.String("values") != "" {
+		values, err = parseFileToString(ctx.String("values"))
```

Comment:
is it possible to extract this logic in a separate function and call that here and in templateInstall as it seems common to both?